### ISSUE TITLE
Swift Package Manager Support

### ISSUE THREAD

**Issue by schlingding** [dji-sdk/Mobile-SDK-iOS#401]
We have migrated all of our frameworks to Swift Package Manage for Xcode 12, but DJI libraries currently do NOT support Swift Package Manager.

When can we expect SPM support?

Thank you

**Comment by dji-dev** [comment]
Agent comment from Luce Luo in <a href='https://djisdksupport.zendesk.com/agent/tickets/40069'>Zendesk ticket #40069</a>:<br><br>Dear Customer,

Thank you for contacting DJI.
The MSDK use the CocoaPods to implement the project, please follow this integration documentation.
https://developer.dji.com/mobile-sdk/documentation/application-development-workflow/workflow-integrate.html#xcode-project-integration

Thanks,

Luce Luo
DJI Developer Support

**Comment by schlingding** [comment]
As I mentioned in an email:

- CocoaPods is a community and THIRD party solution NOT supported directly in Xcode. CocoaPods has a THIRD party client application installed which is responsible for manipulating the Xcode project, schemes, and workspace.
- Swift Package Manager is a FIRST party solution which is directly integrated in Xcode (Xcode 12 being full support for SPM). FULL SUPPORT being the key.

As a developer of ENTERPRISE Applications, the preference will always be the FIRST party solutions.

**Comment by dji-dev** [comment]
Agent comment from Luce Luo in <a href='https://djisdksupport.zendesk.com/agent/tickets/40069'>Zendesk ticket #40069</a>:<br><br>Dear Customer,

Thank you for contacting DJI.
Sorry that the CocoaPods is an easy way to implement the project for the developer than the SPM.
And so sorry that there is no plan to support the SPM for the being time.

Thanks,

Luce Luo
DJI Developer Support

**Comment by gotev** [comment]
@schlingding I totally agree with you and I'm also migrating everything to Swift Package Manager at work. **When Apple promotes an official way of doing something, suddenly all the other options die, for good**. Recently the same happened with RxSwift being put aside by Combine. Finally package management on iOS has an official name: SPM!

It will be no surprise if Apple decides to do something to discourage or completely deprecate both Carthage and CocoaPods in the upcoming Xcode releases. Both tools allowed us to manage dependencies in the past years, but it's time to move forward. 

For the DJI developers and support guys reading, this article may be useful to start thinking about SPM support in the SDK: https://medium.com/@joesusnick/swift-package-manager-with-a-mixed-swift-and-objective-c-project-part-2-2-e71dad234e6

From a company which does such great products, I expect nothing less. You rock guys, and I'm sure you have all the expertise needed to achieve this 🎉

**Comment by schlingding** [comment]
Any update on SPM support?

**Comment by nab0y4enko** [comment]
Hello! DJI SDK is the last library in our project without SPM support. (We have 40+ dependencies)
Do you plan to add support of native dependency manager?
